developer-hub/src/reference_docs/extrinsic_encoding.rs

```diff
@@ -0,0 +1,156 @@
+//! # Constructing and Signing Extrinsics
+//!
+//! Substrate is configurable enough that extrinsics can take any format, in theory. In practice,
+//! runtimes tend to use our [`sp_runtime::generic::UncheckedExtrinsic`] type to represent extrinsics.
+//! In Polkadot, this is configured
+//! [here](https://github.com/polkadot-fellows/runtimes/blob/94b2798b69ba6779764e20a50f056e48db78ebef/relay/polkadot/src/lib.rs#L1478)
+//! at the time of writing.
+//!
+//! What follows is a description of how extrinsics based on this type are encoded into bytes. These
+//! bytes can then be submitted to a chain for inclusion in a block (this is how we make changes to
+//! the state of a chain), and are ultimately stored in the body of a block.
+//!
+//! # Encoding an Extrinsic
+//!
+//! At a high level, all extrinsics are formed from concatenating some details together, ie:
+//!
+//! ```text
+//! extrinsic_bytes = concat(
+//!     compact_encoded_length,
+//!     version_and_maybe_signature,
+//!     call_data
+//! )
+//! ```
+//!
+//! Let's look at how each of these details is constructed.
+//!
+//! ## compact_encoded_length
+//!
+//! This is a SCALE compact encoded integer which is equal to the length, in bytes, of the rest of
+//! the extrinsic details.
+//!
+//! To obtain this value, we must encode and concatenate together the rest of the extrinsic details
+//! first, and then obtain the byte length of these. We can then compact encode that length, and
+//! prepend it to the rest of the details.
+//!
+//! ## version_and_maybe_signature
+//!
+//! If the extrinsic is _unsigned_, then `version_and_maybe_signature` will be just one byte
+//! denoting the _transaction protocol version_, which is 4.
+//!
+//! If the extrinsic is _signed_ (all extrinsics submitted from users must be signed), then
+//! `version_and_maybe_signature` is obtained by concatenating some details together, ie:
+//!
+//! ```text
+//! version_and_maybe_signature = concat(
+//!     version_and_signed,
+//!     from_address,
+//!     signature,
+//!     signed_extensions_extra,
+//! )
+//! ```
+//!
+//! Each of the details to be concatenated together is explained below:
+//!
+//! ### version_and_signed
+//!
+//! This is one byte, equal to `0x84` or `0b1000_0100` (ie an upper 1 bit to denote that it is
+//! signed, and then the transaction version, 4, in the lower bits).
+//!
+//! ### from_address
+//!
+//! This is the SCALE encoded address of the sender of the extrinsic. This address must correspond
+//! to the account ID (public key) whose private key will be used to sign the extrinsic. The exact
+//! type of the address can vary across different chains.
```

Comment:
Would be good to exactly say that this address is the nth generic of the `UncheckedExtrinsic`.